add NEC UNIVERGE IX2105 IX2207 IX3110

nxzz opened the pull request:
This pull request adds more NEC UNIVERGE IX series routers (IX2105 IX2207 IX3110).

The product datasheet can be found below.
I couldn't find an English datasheet, so it's only a Japanese datasheet.
https://jpn.nec.com/univerge/ix/Info/ix2105.html
https://jpn.nec.com/univerge/ix/Spec/hw-spec.html#ix2207
https://jpn.nec.com/univerge/ix/Spec/hw-spec-ix3000.html#ix3110

danner26 commented on device-types/NEC/UNIVERGE-IX2105.yaml:
```diff
@@ -0,0 +1,31 @@
+comments: |
+  NEC UNIVERGE IX2105, VPN-Compatible High speed Access Router, 7W
+  (1) RJ-45 Console Port, (1) Ethernet 1000BASE-T, (4) Ethernet 1000BASE-T in Switching Hub
+
```
Please remove the extra newline

nxzz replied:
removed.

danner26 commented on device-types/NEC/UNIVERGE-IX2105.yaml:
```diff
@@ -0,0 +1,31 @@
+interfaces:
+  - name: GigaEthernet0
+    type: 1000base-t
+  - name: GigaEthernet1 port 1
```
Are these actually referred to as `GigaEthernet1 port 1`? You should define these as they appear in the config.. If this is correct, please let me know

nxzz replied:
`GigaEthernet1 port 1` is the actual interface name.
Here are the results of `show devices`.

```
ix2105# show devices
Device GigaEthernet0 is up
  ifIndex is 1
  GigaEthernet status:
    Physical address is 00:60:b9:xx:xx:xx
    Port status is up
    Full-duplex, 1G b/s, 1000BaseT, MDI
....
Device GigaEthernet1 is up
  ifIndex is 2
  GigaEthernet status:
    Physical address is 00:60:b9:xx:xx:xx
    Promiscuous mode is disabled
    Extended card is switching hub
      Port 1 status is up
        Full-duplex, 1G b/s, 1000BaseT, MDI-X
        VLAN-ID N/A, LAG-ID N/A
        0 CRC errors, 0 collisions
      Port 2 status is down
        Full-duplex, 1G b/s, 1000BaseT, MDI-X
        VLAN-ID N/A, LAG-ID N/A
        0 CRC errors, 0 collisions
      Port 3 status is down
        Full-duplex, 1G b/s, 1000BaseT, MDI-X
        VLAN-ID N/A, LAG-ID N/A
        0 CRC errors, 0 collisions
      Port 4 status is down
        Full-duplex, 1G b/s, 1000BaseT, MDI-X
        VLAN-ID N/A, LAG-ID N/A
        0 CRC errors, 0 collisions
    Entered multicast address 0 peak, 0 current
    Entered vrrp address 0 peak, 0 current
    Statistics:
....
```

danner26 commented on device-types/NEC/UNIVERGE-IX2105.yaml:
```diff
@@ -0,0 +1,31 @@
+    type: 1000base-t
+  - name: GigaEthernet1 port 1
+    type: 1000base-t
+  - name: GigaEthernet1 port 2
```
Are these actually referred to as `GigaEthernet1 port 2`? You should define these as they appear in the config.. If this is correct, please let me know

nxzz replied:
`GigaEthernet1 port 2` is the actual interface name.
Here are the results of `show devices`.

```
ix2105# show devices
Device GigaEthernet0 is up
  ifIndex is 1
  GigaEthernet status:
    Physical address is 00:60:b9:xx:xx:xx
    Port status is up
    Full-duplex, 1G b/s, 1000BaseT, MDI
....
Device GigaEthernet1 is up
  ifIndex is 2
  GigaEthernet status:
    Physical address is 00:60:b9:xx:xx:xx
    Promiscuous mode is disabled
    Extended card is switching hub
      Port 1 status is up
        Full-duplex, 1G b/s, 1000BaseT, MDI-X
        VLAN-ID N/A, LAG-ID N/A
        0 CRC errors, 0 collisions
      Port 2 status is down
        Full-duplex, 1G b/s, 1000BaseT, MDI-X
        VLAN-ID N/A, LAG-ID N/A
        0 CRC errors, 0 collisions
      Port 3 status is down
        Full-duplex, 1G b/s, 1000BaseT, MDI-X
        VLAN-ID N/A, LAG-ID N/A
        0 CRC errors, 0 collisions
      Port 4 status is down
        Full-duplex, 1G b/s, 1000BaseT, MDI-X
        VLAN-ID N/A, LAG-ID N/A
        0 CRC errors, 0 collisions
    Entered multicast address 0 peak, 0 current
    Entered vrrp address 0 peak, 0 current
    Statistics:
....
```

danner26 commented on device-types/NEC/UNIVERGE-IX2105.yaml:
```diff
@@ -0,0 +1,31 @@
+    type: 1000base-t
+  - name: GigaEthernet1 port 2
+    type: 1000base-t
+  - name: GigaEthernet1 port 3
```
Are these actually referred to as `GigaEthernet1 port 3`? You should define these as they appear in the config.. If this is correct, please let me know

nxzz replied:
`GigaEthernet1 port 3` is the actual interface name.
Here are the results of `show devices`.

```
ix2105# show devices
Device GigaEthernet0 is up
  ifIndex is 1
  GigaEthernet status:
    Physical address is 00:60:b9:xx:xx:xx
    Port status is up
    Full-duplex, 1G b/s, 1000BaseT, MDI
....
Device GigaEthernet1 is up
  ifIndex is 2
  GigaEthernet status:
    Physical address is 00:60:b9:xx:xx:xx
    Promiscuous mode is disabled
    Extended card is switching hub
      Port 1 status is up
        Full-duplex, 1G b/s, 1000BaseT, MDI-X
        VLAN-ID N/A, LAG-ID N/A
        0 CRC errors, 0 collisions
      Port 2 status is down
        Full-duplex, 1G b/s, 1000BaseT, MDI-X
        VLAN-ID N/A, LAG-ID N/A
        0 CRC errors, 0 collisions
      Port 3 status is down
        Full-duplex, 1G b/s, 1000BaseT, MDI-X
        VLAN-ID N/A, LAG-ID N/A
        0 CRC errors, 0 collisions
      Port 4 status is down
        Full-duplex, 1G b/s, 1000BaseT, MDI-X
        VLAN-ID N/A, LAG-ID N/A
        0 CRC errors, 0 collisions
    Entered multicast address 0 peak, 0 current
    Entered vrrp address 0 peak, 0 current
    Statistics:
....
```

danner26 commented on device-types/NEC/UNIVERGE-IX2105.yaml:
```diff
@@ -0,0 +1,31 @@
+    type: 1000base-t
+  - name: GigaEthernet1 port 3
+    type: 1000base-t
+  - name: GigaEthernet1 port 4
```
Are these actually referred to as `GigaEthernet1 port 4`? You should define these as they appear in the config.. If this is correct, please let me know

nxzz replied:
`GigaEthernet1 port 4` is the actual interface name.
Here are the results of `show devices`.

```
ix2105# show devices
Device GigaEthernet0 is up
  ifIndex is 1
  GigaEthernet status:
    Physical address is 00:60:b9:xx:xx:xx
    Port status is up
    Full-duplex, 1G b/s, 1000BaseT, MDI
....
Device GigaEthernet1 is up
  ifIndex is 2
  GigaEthernet status:
    Physical address is 00:60:b9:xx:xx:xx
    Promiscuous mode is disabled
    Extended card is switching hub
      Port 1 status is up
        Full-duplex, 1G b/s, 1000BaseT, MDI-X
        VLAN-ID N/A, LAG-ID N/A
        0 CRC errors, 0 collisions
      Port 2 status is down
        Full-duplex, 1G b/s, 1000BaseT, MDI-X
        VLAN-ID N/A, LAG-ID N/A
        0 CRC errors, 0 collisions
      Port 3 status is down
        Full-duplex, 1G b/s, 1000BaseT, MDI-X
        VLAN-ID N/A, LAG-ID N/A
        0 CRC errors, 0 collisions
      Port 4 status is down
        Full-duplex, 1G b/s, 1000BaseT, MDI-X
        VLAN-ID N/A, LAG-ID N/A
        0 CRC errors, 0 collisions
    Entered multicast address 0 peak, 0 current
    Entered vrrp address 0 peak, 0 current
    Statistics:
....
```

danner26 commented on device-types/NEC/UNIVERGE-IX2207.yaml:
```diff
@@ -0,0 +1,37 @@
+comments: |
+  NEC UNIVERGE IX2207, VPN-Compatible High speed Access Router, 24W
+  (1) RJ-45 Console Port, (2) Ethernet 1000BASE-T, (4) Ethernet 1000BASE-T in Switching Hub, (2) USB2.0 Type-A
+
```
Please remove the extra newline

nxzz replied:
removed.

danner26 commented on device-types/NEC/UNIVERGE-IX2207.yaml:
```diff
@@ -0,0 +1,37 @@
+    type: 1000base-t
+  - name: GigaEthernet1
+    type: 1000base-t
+  - name: GigaEthernet2 port 1
```
Are these actually referred to as `GigaEthernet2 port 1`? You should define these as they appear in the config.. If this is correct, please let me know
A sample dump of the config would be appreciated as well, just to make sure this is correct

nxzz replied:
GigaEthernet2 port 1 is the actual interface name.
Here are the results of show devices.
```
IX2207# show devices
Device GigaEthernet0 is up
  ifIndex is 1
  GigaEthernet status:
    Physical address is 00:60:b9:xx:xx:xx
    Port status is up
    Full-duplex, 1G b/s, 1000BaseT, MDI
    Promiscuous mode is disabled
    Entered multicast address 4 peak, 4 current
    Entered vrrp address 0 peak, 0 current
    Statistics:
....
Device GigaEthernet1 is down
  ifIndex is 2
  GigaEthernet status:
    Physical address is 00:60:b9:xx:xx:xx
    Port status is down
    Full-duplex, 1G b/s, 1000BaseT, MDI
    Promiscuous mode is disabled
    Entered multicast address 0 peak, 0 current
    Entered vrrp address 0 peak, 0 current
    Statistics:
....
Device GigaEthernet2 is up
  ifIndex is 3
  GigaEthernet status:
    Physical address is 00:60:b9:xx:xx:xx
    Promiscuous mode is disabled
    Extended card is switching hub
      Port 1 status is up
        Full-duplex, 1G b/s, 1000BaseT, MDI-X
        VLAN-ID N/A, LAG-ID N/A
        0 CRC errors, 0 collisions
      Port 2 status is down
        Full-duplex, 1G b/s, 1000BaseT, MDI-X
        VLAN-ID N/A, LAG-ID N/A
        0 CRC errors, 0 collisions
      Port 3 status is up
        Full-duplex, 1G b/s, 1000BaseT, MDI-X
        VLAN-ID N/A, LAG-ID N/A
        0 CRC errors, 0 collisions
      Port 4 status is down
        Full-duplex, 1G b/s, 1000BaseT, MDI-X
        VLAN-ID N/A, LAG-ID N/A
        0 CRC errors, 0 collisions
    Entered multicast address 0 peak, 0 current
    Entered vrrp address 0 peak, 0 current
    Statistics:
....
```

danner26 commented on device-types/NEC/UNIVERGE-IX2207.yaml:
```diff
@@ -0,0 +1,37 @@
+    type: 1000base-t
+  - name: GigaEthernet2 port 4
+    type: 1000base-t
+  - name: USB0
```
Are these USB ports just to transfer data from a USB? If so, please remove these as they are not meant to be modelled like this

nxzz replied:
Data transfer is also possible, but the main purpose is to connect a USB LTE modem.

danner26 commented on device-types/NEC/UNIVERGE-IX3110.yaml:
```diff
@@ -0,0 +1,34 @@
+  NEC UNIVERGE IX3110, VPN-Compatible High speed Access Router, 40W
+  (1) RJ-45 Console Port, (4) Ethernet 1000BASE-T, (4) Ethernet 1000BASE-X-SFP
+  Every Ethernet port is a 1000BASE-T/SFP combo port
+
```
Please remove the extra newline

nxzz replied:
removed.

danner26 reviewed (changes requested):
Please review my questions. Are the ports actually referred to in the config as `GigaEthernet 1 port 1`? Or are they more like the UNIVERGE IX3110 `GigaEthernet1`, `GigaEthernet2` and so on?

If they are in fact like you have them defined, a sample output of the device config would be appreciated.. Thanks!

nxzz commented:
Thank you for your review.

In the NEC IX series, the interface that operates as a switching hub is called "GigaEthernetX port Y".
Otherwise, GigaEthernetX only.

danner26 commented:
Thanks for all of your work @nxzz 
One last thing, for `USB0` I see that you mentioned they are mainly for USB LTE additions. How do you plan on using this in NetBox?

nxzz commented:
I would like to define an LTE modem as a device ,
and show that the LTE modem is connected to router.
If this is not a good way to indicate an LTE modem is connected, I would remove the USB.

danner26 commented:
I am discussing internally with the maintainers on how to best handle this, feel free to ping me if you dont hear back in a few days

nxzz commented:
Hi @danner26 .
is there any progress?

danner26 commented:
Hey @nxzz still working on this, the team is very busy at the moment and unfortunately this has fallen down the list. I will try to get it some more attention ASAP

Reach back out in another week or two if you dont hear back.. I am sorry

danner26 commented:
Hey @nxzz while I was cleaning up some other devices I realized that there is an interface type `lte`. I saw other devices are using `lte` as the interface type and calling the port USB0 or something like that. Would that work for your implementation? If so I will double check with the other maintainers and make sure it works for everyone.

nxzz commented:
Added weight to the definition.
Changed USB0/USB1 type to lte.

@danner26 
For my purposes, no network devices other than LTE modems are expected to be connected to USB0/USB1 on the IX series, so setting the type to lte solves this problem.

danner26 commented on device-types/NEC/UNIVERGE-IX2207.yaml:
```diff
@@ -0,0 +1,38 @@
+---
+manufacturer: NEC
+model: UNIVERGE IX2207
+slug: univerge-ix2207
```
Please append `nec-` to the slug

danner26 commented on device-types/NEC/UNIVERGE-IX3110.yaml:
```diff
@@ -0,0 +1,35 @@
+---
+manufacturer: NEC
+model: UNIVERGE IX3110
+slug: univerge-ix3110
```
Please append `nec-` to the slug

nxzz replied:
Added.

danner26 commented on device-types/NEC/UNIVERGE-IX3110.yaml:
```diff
@@ -0,0 +1,35 @@
+  - name: GigaEthernet3
+    type: 1000base-t
+power-ports:
+  - name: Module 0
```
It looks like the PSUs are modular, is this the case? If so, please remove these definitions and create them under `module-bays`. Then create the applicable PSUs in the module-types folder if they do not already exist

nxzz replied:
Add PSU BE000220 to ./module-types/NEC/BE000220.yaml and
Fixed definition of UNIVERGE IX3110.
